### DISCUSSION TITLE
permit <host> does not match ACE pattern

### DISCUSSION THREAD

**Discussion by mattr64** [vladimirs-git/cisco-acl#4] [Q&A]
Hey Vlad,

First of all your package is awesome! :) Thank you for developing this.

I have a question, maybe I'm not invoking the package correctly but if I use an ACL with just a '`permit <host>`' syntax from an IOS-XE config, I get an error.

Example config:

```
config = """
ip access-list standard ACL-test-1
 permit 10.1.1.1
"""
```

Then attempting a simple acl.line print:

```
acls = cisco_acl.acls(config=config, platform="ios", )
acl = acls[0]
print(acl.line, "\n")
```

I get the error:
`WARNING:root:ValueError: invalid line='permit 10.1.1.1'. line='permit 10.1.1.1' does not match ACE pattern
`
Is this expected? I could work around this by adding '0.0.0.0' as the mask to the end of each line, but would be great if the parser could handle this type of input.

If I add the mask to the end of each line, it does work as expected.

Edit: I've got a workaround where we add the 0.0.0.0 mask to the end of any explicit permit/deny without a mask:

```
def add_mask_to_acl(config):
    # Regular expression to match permit/deny lines without a mask
    pattern = r"^(permit|deny)\s+(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})$"
    new_config_lines = []

    for line in config.split('\n'):
        match = re.match(pattern, line.strip())
        if match:
            modified_line = f" {match.group(1)} {match.group(2)} 0.0.0.0"
            new_config_lines.append(modified_line)
        else:
            new_config_lines.append(line)
    return "\n".join(new_config_lines)
```

```
modified_config = add_mask_to_acl(config)
acls = cisco_acl.acls(config=modified_config, platform="ios")
acl = acls[0]
print(acl.line, "\n")
```

**Comment by vladimirs-git** [comment]
Thank you for your interest in this project!
Unfortunately, I haven't worked with this package for standard ACLs before, and that's the reason for the mistakes.
In version [3.2.3](https://github.com/vladimirs-git/cisco-acl/releases/tag/3.2.3), I tried to implement standard ACL for correct host syntax, but the output still includes the `host` keyword.
```python
from cisco_acl import Acl

lines1 = """
ip access-list standard ACL1
  permit 10.0.0.1
  permit host 10.0.0.2
  permit 10.0.0.3 0.0.0.0
"""

# cisco-acl==3.2.3
acl1 = Acl(lines1)
print(str(acl1))
# ip access-list standard ACL1
#   permit host 10.0.0.1
#   permit host 10.0.0.2
#   permit host 10.0.0.3

# todo
# ip access-list standard ACL1
#   permit 10.0.0.1
#   permit 10.0.0.2
#   permit 10.0.0.3
```

Currently, I don't have enough time to fix this issue fully. I will fix it when I have a project related to standard ACLs.

The best thanks is to star this project.
